shub_image/deploy.py

```diff
@@ -48,17 +48,19 @@
 @click.option("--username", help="docker registry name")
 @click.option("--password", help="docker registry password")
 @click.option("--email", help="docker registry email")
+@click.option("--insecure", is_flag=True, help="use insecure registry")
 @click.option("--async", is_flag=True, help="enable asynchronous mode")
-def cli(target, debug, version, username, password, email, async):
-    deploy_cmd(target, version, username, password, email, async)
+def cli(target, debug, version, username, password, email, insecure, async):
+    deploy_cmd(target, version, username, password, email, insecure, async)
 
 
-def deploy_cmd(target, version, username, password, email, async):
+def deploy_cmd(target, version, username, password, email, insecure, async):
     config = utils.load_release_config()
     project, endpoint, apikey = config.get_target(target)
     image = config.get_image(target)
     version = version or config.get_version()
     image_name = utils.format_image_name(image, version)
+    username = None if insecure else (username or apikey)
```

Comment:
Isn't insecure registry simply [a registry without tls](https://docs.docker.com/registry/insecure/)? Why you set `username = None` here?